app/controllers/works_controller.rb

```diff
@@ -0,0 +1,101 @@
+class WorksController < ApplicationController
+  before_action :find_work, except: [:index, :new, :create]
+
+  def index
+    @works = Work.all
+  end
+
+  def show
+    if @work.nil?
+      render :file => "#{Rails.root}/public/404.html",  layout: false, status: :not_found
+      return
```

Comment:
This works, but it's a bit ugly, better to provide a more attractive 404 page or redirect someplace.